### ISSUE TITLE
Unabel to write in the cache directory

### ISSUE THREAD

**Issue by vjvikram93** [mautic/mautic#4069]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Yes
| Feature request? | No
| Enhancement? | No

## Description:
This is a fresh installation of mautic on a Centos 7 minimal installation. Once the installation is done, the website shows the following error: 

The site is currently offline due to encountering an error. If the problem persists, please contact the system administrator.
System administrators, check server logs for errors.

The logs generated are mentioned below.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | Latest release
| PHP version | PHP 7.0.18 (cli) (built: Apr 15 2017 07:09:11) ( NTS )

### Steps to reproduce:
1. Fresh install on the Centos7 machine with the above mentioned versions.
2. Once the webpage is opened for installation, it throws the errors mentioned in the log below.
 
### Log errors: 
In case the cache directory is deleted:

[Tue May 16 20:07:41.447445 2017] [:error] [pid xxxx] [client xxx.xxx.xxx.xxx:xxxxx] RuntimeException: Unable to create the cache directory (/var/www/html/mautic/app/cache/prod)\n - in file /var/www/html/mautic/app/AppKernel.php - at31

In case the cache directory is manually created with proper permissions

[Tue May 16 20:54:05.980313 2017] [:error] [pid xxxx] [client xxx.xxx.xxx.xxx:xxxxx] RuntimeException: Unable to write in the cache directory (/var/www/html/mautic/app/cache/prod)\n - in file /var/www/html/mautic/app/AppKernel.php -  534

### Apache user
user: apache
group: apache

### File Permissions
##app dir
drwxr-xr-x.  9 apache apache  4096 May 16 20:06 .
drwxr-xr-x.  9 apache apache  4096 May 16 19:51 ..
-rw-r--r--.  1 apache apache   141 Apr 26 03:18 AppCache.php
-rw-r--r--.  1 apache apache 18388 Apr 26 03:18 AppKernel.php
-rw-r--r--.  1 apache apache   383 Apr 26 03:18 autoload.php
-rw-r--r--.  1 apache apache 41038 Apr 26 03:20 bootstrap.php.cache
drwxr-xr-x. 25 apache apache  4096 Apr 26 03:18 bundles
drwxr-xr-x.  3 apache apache  4096 May 16 20:53 cache
drwxr-xr-x.  2 apache apache  4096 Apr 26 03:20 config
-rwxr-xr-x.  1 apache apache  1469 Apr 26 03:18 console
-rw-r--r--.  1 apache apache    13 Apr 26 03:18 .htaccess
drwxr-xr-x.  2 apache apache  4096 Apr 26 03:20 logs
drwxr-xr-x.  4 apache apache  4096 Apr 26 03:18 middlewares
drwxr-xr-x.  2 apache apache  4096 Apr 26 03:18 migrations
drwxr-xr-x.  3 apache apache  4096 Apr 26 03:18 Resources
-rw-r--r--.  1 apache apache     5 Apr 26 03:18 version.txt

##cache dir
drwxr-xr-x. 3 apache apache 4096 May 16 20:53 .
drwxr-xr-x. 9 apache apache 4096 May 16 20:06 ..
drwxr-xr-x. 2 apache apache 4096 May 16 20:53 prod

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._

**Comment by escopecz** [comment]
Hi, I close this since it's not a Mautic issue, but a server issue. Look at https://github.com/mautic/mautic/issues/4044 if it point you the right direction. But the best would be if you could consult it with your server administrator/provider.

**Comment by vjvikram93** [comment]
Hi, I looked into the #4044 ticket that you referred to and have cross verified the permissions. They all seem to be proper. (apache user having all the ownership; directory permissions set to 755 and file permissions to 644; even tried 777 on the cache directory but no use.)

I don't think consulting with the system administrator/provider would be of any use because he also has no idea in this matter. (it's me. I have the root. lol)

Would really appreciate if you could help me out on this.

**Comment by 3ullShark** [comment]
@vjvikram93 not sure if you've solved this yet but it could be because of SELinux if you have all the correct permissions.

**Comment by bartmcpherson** [comment]
This is a documentation issue. A list of directories needing write permission by the web server should be provided as part of the prerequisites.

SELinux changes can then be applied to those directories.

Example: `chcon -t httpd_sys_rw_content_t /var/www/html/cache -R`

**Comment by nartcan** [comment]
I constantly get same error. I tried on vultr and digital ocean, but no use. Please help!

**Comment by datmt** [comment]
you should make www-data the owner of your mautic folder

**Comment by SBK-Sound** [comment]
Also, if you're on shared hosting, and the host decided to upgrade the management panel, or move to a new server etc.... sometimes the user/account home directory gets changed, 

e.g: /home1/your_user_account_name/public_html/your_website/mautic
to    /home4/your_user_acc...

Resulting in a cannot write/create cache/prod error, because the path does not exist!

In which case edit the /mautic/app/config/local.php file and look for the cache path and log path to make sure the top directory matches the server directory structure.
If this happens, be aware that it may be a temporary change (temporary can mean a few days or a few months) and it may change back eventually, meaning you'll have to edit the files again.

I had this on hostmonster, and since they are a subsidiary of an umbrella company (Endurance International Group) which owns the 20 largest host brands (bluehost, hostgator, godaddy, etc... pretty much any of those kind of hosts you see advertised everywhere, yes it's a monopoly) then they most likely run all the subsidiary companies the same way, and consequently have the same server procedures etc... all the same way.

**Comment by mmangione** [comment]
> you should make www-data the owner of your mautic folder

I would just like to say... what the hell? I am using the docker image mautic:v3. I had to get into the image with
`docker exec -it mautic /bin/bash`

Then, as root, I had to run:
`chown -R www-data:www-data /var/www/html`

In the stock Docker image, root owns all of the files and folders in the /var/www/html directory - negating the whole purpose of putting stuff there.

In any event, if anyone else runs into this issue on the Docker image of Mautic, there is your solution.
